### PR TITLE
YALB-78: Pagination

### PR DESCRIPTION
**Pull request by ModulesUnraveled**
### Description of work
- Creates the initial Pagination component
- Creates Event Grid page (should have had this before)
- Creates Event List page including the pager at the bottom

### Testing Link(s)
- [ ] Navigate to the [Component Story](https://deploy-preview-102--dev-component-library-twig.netlify.app/?path=/story/molecules-navigation-pager--basic)

### Functional Review Steps
- [ ] Verify you can install the component with the CLI

### Design Review
- [ ] Verify the designs match the [Figma Designs](https://www.figma.com/file/OVfmCW5s1gI7xnrM8ibX2y/UI-Kit-Alpha-%5BYaleSites%5D)
- [ ] Review any new [Percy builds](https://percy.io/95144ff9/component-library-twig)

### Accessibility Review
- [ ] Verify the component meets Accessibility requirements

### CODE REVIEW

**Comment by ahitson** [comment]
- The active page should skip hover/focus.
- Change the color of the outline and active page number to basic-brown-gray.
- Can we increase the padding between previous/last and the page numbers?
- Is there a way to see mobile?

**Comment by ModulesUnraveled** [comment]
@ahitson I've addressed all of your requests!

**Comment by ahitson** [comment]
- Change "previous" to "back"
- Use the chevron icon instead of the text arrows < >
- On the basic 3/5 mobile there aren't directional controls the directional controls should appear

**Comment by ModulesUnraveled** [comment]
@ahitson Done

**Comment by ahitson** [comment]
- Add size spacing 1 margin between the back/next and icon
- Change the icon size to .8 em
- Make directional arrows always present except when on the 1st (no back) and last pages (no next)

**Review by ahitson — APPROVED** [review]

**Review by kara-franco — CHANGES_REQUESTED** [review]
Excellent!! Three requests to make for a better screen reader user experience: 

- [ ] Add `aria-label="Pagination"`to the `<nav>`
Screen Shot 2022-07-11 at 12 48 00 PM

- [ ] The title attribute + the link name are really redundant for screen reader user, they hear "Link, Page 1, Go to page 1". Refer to the US Web Design System for arial-label example: https://designsystem.digital.gov/components/pagination/ 

Screen Shot 2022-07-11 at 1 23 35 PM

- [ ] Add `aria-current="page" `to current page, so users know where they are. 

Screen Shot 2022-07-11 at 12 57 22 PM

**Review by miketullo95 — APPROVED** [review]

File: components/02-molecules/pager/pager.twig [hunk 1]
```diff
@@ -0,0 +1,79 @@
+{% set pager__base_class = 'pager' %}
```
Comment (callinmullaney) [inline components/02-molecules/pager/pager.twig:1]: This file seems like it should be split into partial at the very least (see main feedback from the PR review). It's using 4 different types of markup with at least 4 levels of embedding said markup. If you decide to not extend the base menu for pager I would probably partial out this file to `pager-list.twig`, `pager-item.twig`, and `pager-link.twig`.

Comment (ModulesUnraveled) [reply components/02-molecules/pager/pager.twig:1]: I think I disagree... this is a very different component from a typical "menu". There are a lot of custom things this needs to support (`items.previous` (and next), `items.first` (and last), and then the "regular" `items.pages` - They're all mixed together in one array - and no other menu has those things.

This one also needs to pass icons into the content of specific items, which I think would be a pain to shoehorn into the base menu component.

There's also the fact that the "desktop" and "mobile" pagers are both here, but each are hidden at relevant screen sizes.

Also, the base menu components add a lot of extra stuff like menu levels - which will never apply to a pager.

On a side - but related - note, I wanted to talk with you about our implementation of the list component. I'm not sure the value is worth the complexity it adds. Let's zoom tomorrow, if you're free.

**Comment by ModulesUnraveled** [comment]
@callinmullaney I'm going to merge this to unblock a release. I scheduled a zoom for us to go over your feedback, and if anything needs to be done as a result, I'll open a followup PR 👍

**Pull request merged**